enterprise/app/trends/drilldown_page.tsx

```diff
@@ -219,16 +226,54 @@ export default class DrilldownPageComponent extends React.Component<Props, State
       .finally(() => this.setState({ loadingDrilldowns: false }));
   }
 
-  fetchInvocationList() {
-    // TODO(jdhollen): Support fetching invocations based on executions data.
-    if (
-      !this.props.user?.selectedGroup ||
-      isExecutionMetric(this.selectedMetric.metric) ||
-      !this.currentHeatmapSelection
-    ) {
+  fetchExecutionList(heatmapSelection: HeatmapSelection) {
+    if (!capabilities.config.executionSearchEnabled) {
       return;
     }
-    this.setState({ loadingInvocations: true, invocationsFailed: false, invocationsData: undefined });
+    this.setState({
+      loadingEvents: true,
+      eventsFailed: false,
+      eventData: undefined,
+    });
+    const filterParams = getProtoFilterParams(this.props.search);
+    let request = new execution_stats.SearchExecutionRequest({
+      query: new execution_stats.ExecutionQuery({
+        invocationHost: filterParams.host,
+        invocationUser: filterParams.user,
+        repoUrl: filterParams.repo,
+        branchName: filterParams.branch,
+        commitSha: filterParams.commit,
+        command: filterParams.command,
+        pattern: filterParams.pattern,
+        role: filterParams.role || [],
+        updatedAfter: filterParams.updatedAfter,
+        updatedBefore: filterParams.updatedBefore,
+        invocationStatus: filterParams.status || [],
+        filter: this.toStatFilterList(heatmapSelection),
+      }),
+      pageToken: "",
+      count: 25,
+    });
+    this.addZoomFiltersToQuery(request.query!);
+
+    rpcService.service
+      .searchExecution(request)
+      .then((response) => {
+        console.log(response);
+        this.setState({
+          eventData: { executions: response.execution },
+        });
+      })
+      .catch(() => this.setState({ eventsFailed: true, eventData: undefined }))
```

Comment:
instead of `eventsFailed` you might want to store the error message and display it to the user so that we can diagnose. Probably the easiest thing is to use `errorService.handleError` to show the error banner instead of having custom error handling code

Reply:
I do like having *some* permanent / visible error state in the specific section that failed, but I don't really want to dump the whole error into it--so let's do both.